download/download.go

```diff
@@ -311,6 +311,23 @@ func (d *Downloader) download(ctx context.Context, m metrics.Metrics) (*download
 			if d.sizeLimitBytes != nil {
 				reader = reader.WithSizeLimitBytes(*d.sizeLimitBytes)
 			}
+
+			if d.logger.GetLevel() <= logging.Debug {
```

Comment:
I'm not sure I understand the `<=` here. Is Info <= Debug? If so, I'd expect a `>=`, so that the message is only logged when the level actually _is_ debug or more. (We don't have a higher level, though, do we? 🤔 )